homework_2/internal/domain/services/feedProcessorService.go

```diff
@@ -3,37 +3,53 @@ package services
 import (
 	"context"
 	"log"
+	"sync"
 
 	"code-cadets-2021/lecture_2/06_offerfeed/internal/domain/models"
 )
 
 type FeedProcessorService struct {
-	feed  Feed
+	feeds  []Feed
 	queue Queue
 }
 
 func NewFeedProcessorService(
-	feed Feed,
+	feeds []Feed,
 	queue Queue,
 ) *FeedProcessorService {
 	return &FeedProcessorService{
-		feed:  feed,
+		feeds:  feeds,
 		queue: queue,
 	}
 }
 
 func (f *FeedProcessorService) Start(ctx context.Context) error {
-	updates := f.feed.GetUpdates()
+	var updates []chan models.Odd
+
+	for _, feed := range f.feeds {
+		updates = append(updates, feed.GetUpdates())
+	}
 	source := f.queue.GetSource()
 
 	defer close(source)
 	defer log.Printf("shutting down %s", f)
 
-	for update := range updates {
-		update.Coefficient *= 2
-		source <- update
+	wg := &sync.WaitGroup{}
+	wg.Add(len(updates))
+
+	for _, ch := range updates {
+		go func(c chan models.Odd) {
```

Comment:
Instead of using goroutines here, you could just as easily try reading form multiple input channels (like in this example - https://gobyexample.com/select). It should behave the same, but it would be slightly simpler (no need for the wait group) - so feel free to leave your solution as-is.

Also, do you need to check whether the context was cancelled here?

Reply:
I wanted my implementation to work for N channels and I didn't know how to do that dynamically with select statements as looping through them wouldn't work.

Reply:
Makes sense!